doc/working-with-git-and-github.md

```diff
@@ -0,0 +1,202 @@
+to Numberscope in the same document.
+
+Contents
+
+-   [Basic operations](#basic-operations)
+    -   [Add changes](#add-changes)
+    -   [Commit changes](#commit-changes)
+    -   [Push changes](#push-changes)
+-   [Advanced operations](#advanced-operations)
+    -   [Stash your changes](#stash-your-changes)
+    -   [Unstash your changes](#unstash-your-changes)
+    -   [Create a branch](#create-a-branch)
+    -   [Push a branch](#push-a-branch)
+    -   [Create a fork](#create-a-fork)
+    -   [Clone a repo](#clone-a-repo)
+    -   [Add a remote](#add-a-remote)
+    -   [Sync local fork with remote original](#sync-local-fork-with-remote-original)
+
+## Basic operations
+
+When using Git, it can be helpful to think of it as photographer who stages
+your work ("adds" changes), takes a snapshot of it ("commits" changes), and
+stores the photo somewhere safe ("pushes" changes to e.g. GitHub).
+
+### Add changes
+
+To add or "stage" your work for a commit or "snapshot", issue the following
+command:
+
+```sh
+git add /path/to/file/here
+```
+
+(You should do this command for each file you need to stage.)
+
+### Commit changes
+
+To commit your work or take a "snapshot" of the work, issue the following
+command:
+
+```sh
+git commit -m "put your commit message here"
+```
+
+It's best to write your commit message in the
+[imperative](https://en.wikipedia.org/wiki/Imperative_mood) and keep it
+shorter than 50 characters.
+
+### Push changes
+
+Before you are able to do this, you either need to install the
+[GitHub command line interface (CLI)](https://cli.github.com/) and log in
+using the GitHub CLI or
+[install an SSH key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account).
+
+To push your changes to a remote version of your repository (i.e. to the
+version on GitHub) or to "store" your photo, issue the following command:
+
+```sh
+git push
+```
+
+## Advanced operations
+
+### Stash your changes
+
+To "stash" your changes (i.e. squirrel them away for later use), issue the
+following command:
+
+```sh
+git stash
+```
+
+### Unstash your changes
+
+To unstash your changes, issue the following command:
+
+```sh
+git stash apply
+```
+
+The above command keeps the changes in your stash. If you wish to discard
+those changes from the stash, issue the following command:
+
+```sh
+git stash pop
+```
+
+### Create a branch
+
+To create a branch and switch to it in one command, issue the following
+command:
+
+```sh
+git checkout -b your_branch_name_here
+```
+
+At Numberscope we use `snake_case` for branch names.
+
+### Push a branch
+
+To create a branch in a remote Git repository and push your changes to that
+branch, issue the following command:
+
+```sh
+git push -u name_of_remote name_of_branch
+```
+
+The name of the remote is most likely `origin` if you are pushing to a fork of
+one of the Numberscope repositories.
```

Comment:
Add something about how you can view your remote names to double check (you told me that command on slack).

Reply:
Oh, I see you do something like that below, maybe you should reorganize / link so that info is available here and there somehow.

Reply:
Added section on checking remotes in https://github.com/numberscope/frontscope/pull/171/commits/4d7ced0c17f1722459205d1bfac4b4dd8812548f.